File: CIVITAS/src/civitas/Dado.java
/*
 * To change this license header, choose License Headers in Project Properties.
 * To change this template file, choose Tools | Templates
 * and open the template in the editor.
 */
package civitas;

import java.util.Random;
/**
 *
 * @author Usuario
 */
 class Dado {
    static final private Dado instance = new Dado();
    static  final private int SalidaCarcel = 5;
    
    static public Dado getInstance(){
        return instance;
    }
    
    private Random random;
    private int ultimoResultado;
    private Boolean debug;
    
    private Dado (){
        random = new Random();
        ultimoResultado = -1;
        debug = false;
        
    }
    
    int tirar(){
       
        if(debug){
            ultimoResultado = 1;
        }else{
             ultimoResultado = (int)(random.nextInt(6))+1;
        }
       
        return ultimoResultado;
    }
    
    Boolean salgoDeLaCarcel(){
        int d = this.tirar();
        if(SalidaCarcel == d)
            return true;
      
        else
            return false;
    }
    
    int quienEmpieza(int n){
        return random.nextInt(n);
    }
    
    void setDebug(Boolean d){
        debug = d;
        
        Diario.getInstance().ocurreEvento("Se ha puesto el modo debug a "+d);
    }
    
    int getUltimoResultado(){
        return this.ultimoResultado;
    }
}

File: CIVITASRUBY/lib/dado.rb
# To change this license header, choose License Headers in Project Properties.
# To change this template file, choose Tools | Templates
# and open the template in the editor.
require "singleton"
require_relative "diario"

module Civitas
  
  class Dado
    include Singleton
    attr_reader :ultimo_resultado
    @@SALIDA_CARCEL = 5
    def initialize
      @ultimo_resultado = -1
      @debug = false
    end
    
    def tirar
      if @debug == true
        @ultimo_resultado = 1
      else
        @ultimo_resultado = rand(6)+1
      end
      return @ultimo_resultado
    end
    
    def salgo_de_la_carcel
      d = tirar
      if(@@SALIDA_CARCEL == d)
        return true;
      else
        return false;
      end
    end
    
    def quien_empieza(n)
      return rand(n)
    end
    
    def set_debug (d)
      @debug = d
      Diario.instance.ocurre_evento("se ha puesto el modo debug a "+ d.to_s)
    end
  end
end